api/maven-api-core/src/main/java/org/apache/maven/api/Artifact.java

```diff
@@ -23,76 +23,83 @@
 import org.apache.maven.api.annotations.Nonnull;
 
 /**
- * An artifact points to a resource such as a jar file or war application.
+ * Pointer to a resolved resource such as a <abbr>JAR</abbr> file or <abbr>WAE</abbr> application.
+ * {@code Artifact} instances are created when <dfn>resolving</dfn> {@link Artifact} instances.
+ * Resolving is the process that selects a {@linkplain #getVersion() particular version}
+ * and downloads the artifact in the local repository.
```

Comment:
So things are a bit blurry:
 * artifact resolution is:  resolve version (see `VersionResolver#resolveVersion()`), and then resolve (i.e. download) the file
 * dependency resolution is: collect dependencies, flatten the graph, resolve (i.e. download) the file

During dependency collection, versions are resolved (@cstamas right?). The flattening phase removes branches of the graph so that one artifact per GA is present. The resolution phase will then download the result artifacts.

We don't have any way to download an artifact for which the version has been resolved though (that may be a missing bit).  So if we introduce `ResolvedArtifact` and `ResolvedDependency`, it may help a bit:
  * version resolution: ArtifactCoordinate -> Artifact
  * artifact resolution: Artifact -> ResolvedArtifact
  * dependency collection -> Node + Dependency
  * dependency resolution: list of Dependency -> list of ResolvedDependency

Reply:
Added a "definitions of terms" section in `package-info.java` with the above explanation about artifact resolution and dependency resolution. Since those definitions include the download part, they answer my [previous comment](https://github.com/apache/maven/pull/1640#discussion_r1714881957).